### ISSUE TITLE
Read COG tile

### ISSUE THREAD

**Issue by geospatial-jeff** [geospatial-jeff/async-cog-reader#3]
Once #2 is ready to go, we need a method to use IFD/tag metadata to read a given tile.  COGDumper uses `cogdumper.cog_tiles.COGTiff.read_tile` which looks to just a single XYZ tile based on the tile's coordinate with respect to the (top left?) of the image from the appropriate overview.

As @vincentsarago pointed out, we could use pyproj to:
> * get geospatial info from the COG
> * fetch only the internal tile (and overview tiles) for a specific `.read` request.

I think it would be nice to implement something similar to `rasterio.windows` where we can use pyproj to map a particular bounding box to the corresponding XYZ tiles in the COG, but I'm definitely open to other ideas.  This brings some questions.
- How this will work with `rio-tiler-v2` -- if at all.  If the `COGTiff` class can implement a similar interface to `rasterio.io.DatasetReader` it could be passed in as the `src_dst` to `rio_tiler.reader._read` but I'm not sure if that is feasible.

**Comment by vincentsarago** [comment]
> How this will work with rio-tiler-v2 -- if at all. If the COGTiff class can implement a similar interface to rasterio.io.DatasetReader it could be passed in as the src_dst to rio_tiler.reader._read but I'm not sure if that is feasible.

Oh I didn't though about this. To be honest I don't think we should worry about rio-tiler. As rasterio, rio-tiler tries to be able to work with all data, while cog-read should just focus on doing fast partial reading IMO.  `rio-tiler` is using warpedVRT to read the overviews and perform boundless read, I think we could do something realy simple withing cog-reader to achieave the same goal. 

```
async with cog_reader.open("my_cog.tif") as src_dst:
    bounds: Sequence[float] <---- should be in src_dst projection 
    window = src_dst.window_from_bounds(*bounds) <---- https://github.com/mapbox/rasterio/blob/master/rasterio/windows.py#L47-L75

    arr = await src_dst.read(window=window, out_shape=(3, 256, 256) ...)
```
- window read will trigger internal tile reading,
- out_shape will trigger which overview level to read
- basically we will fetch more data that needed and then we will need to resample the array to our out_shape ..... is this possible in python ?
